File: frontend_refactor/src/dtos/category/response/category-tree.response.dto.ts
export interface CategoryTreeResponseDto {
  categoryId: number;
  categoryName: string;
  categoryLevel: number;
  categoryType: 'DOMESTIC' | 'FOREIGN';
  parentCategoryId?: number;
  categoryOrder?: number;
  isActive: boolean;
  discountPolicyId?: number;
  subCategories?: CategoryTreeResponseDto[];
}

File: backend/src/main/java/com/bookhub/bookhub_back/dto/category/response/CategoryTreeResponseDto.java
package com.bookhub.bookhub_back.dto.category.response;

import com.bookhub.bookhub_back.common.enums.CategoryType;
import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CategoryTreeResponseDto {
    private Long categoryId;
    private String categoryName;
    private int categoryLevel;
    private CategoryType categoryType;
    private int categoryOrder;
    private Boolean isActive;
    private Long parentCategoryId;
    private Long discountPolicyId;
    private List<CategoryTreeResponseDto> subCategories;
}
